pkg/web/macaron.go

```diff
@@ -139,6 +139,17 @@ func mwFromHandler(handler Handler) Middleware {
 	}
 }
 
+// a convenience function that is provided for users of contexthandler package (standalone apiservers)
+// who have an implicit dependency on Macron in context but don't want to take a dependency on
+// router additionally
+func EmptyMacronMiddleware(next http.Handler) http.Handler {
```

Comment:
Not using such a middleware makes contexthandler crash because it assumes that the router has completed the initial setup of the macron context key in context. But most of those things such as Req are nil pointers without `createContext`.